database/skylink.go

```diff
@@ -113,20 +113,20 @@ func (db *DB) MarkUnpinned(ctx context.Context, skylink skymodules.Skylink) erro
 	return err
 }
 
-// AddServerForSkylink adds a new server to the list of servers known to be pinning
-// this skylink. If the skylink does not already exist in the database it will
-// be inserted. This operation is idempotent.
+// AddServerForSkylinks adds a new server to the list of servers known to be
+// pinning these skylinks. If a skylink does not already exist in the database
+// it will be inserted. This operation is idempotent.
 //
-// The `markPinned` flag sets the `unpin` field of the skylink to false when
+// The `markPinned` flag sets the `unpin` field of a skylink to false when
 // raised but it doesn't set it to false when not raised. The reason for that is
 // that it accommodates a specific use case - adding a server to the list of
 // pinners of a given skylink will set the unpin field to false is we are doing
 // that because we know that a user is pinning it but not so if we are running
 // a server sweep and documenting which skylinks are pinned by this server.
-func (db *DB) AddServerForSkylink(ctx context.Context, skylink skymodules.Skylink, server string, markPinned bool) error {
-	db.staticLogger.Tracef("Entering AddServerForSkylink. Skylink: '%s', server: '%s'", skylink, server)
-	defer db.staticLogger.Tracef("Exiting  AddServerForSkylink. Skylink: '%s', server: '%s'", skylink, server)
-	filter := bson.M{"skylink": skylink.String()}
+func (db *DB) AddServerForSkylinks(ctx context.Context, skylinks []string, server string, markPinned bool) error {
```

Comment:
I generally agree with the idea and I would have kept these as `skymodules.Skylink` if I could. But the way we call this method is like this:
```go
	unknown, missing := s.staticSkydClient.DiffPinnedSkylinks(dbSkylinks)
	// Add all missing skylinks to the database.
	err = s.staticDB.AddServerForSkylinks(ctx, missing, s.staticServerName, false)
```
and `DiffPinnedSkylinks` operates with strings because it works with data coming from the database. So, if we decide that this is important we can convert the strings to skylinks somewhere for validation purposes but I'd rather do it at the points where those enter the system, i.e. in the sweeper and in the handlers. We are already doing it in the handlers and we're not doing it while sweeping because we trust the contents of the files in the renter's dir. I can definitely add a check there, if that would be helpful.

Reply:
a simple check that all the strings are valid siapaths could be helpful. That would catch data corruption on disk or some other misconfiguration, like somehow looking at the wrong directory. But like you said, maybe in the handler and not here. 

either way, fine for a follow up, or if it is more effort then help we can just dismiss this.

Reply:
I can add it i #41.